### DISCUSSION TITLE
Sortable relation column

### DISCUSSION THREAD

**Discussion by viezel** [filamentphp/filament#83] [Ideas]
Lets use your [demo app](https://github.com/laravel-filament/demo) as an example of this feature. We want to add the number of Orders to the Customer Resource table. We also want this relation column to be sortable.

Today, I can use `formatUsing` or `getValueUsing`

```php
Columns\Text::make('orders')
    ->formatUsing(fn ($record) => count($record)),
```

But none of them are sortable. 

I propose to build a Number column, that requires a number and that would be sortable

```php
Columns\Number::make('orders')
    ->formatUsing(fn ($record) => count($record))
    ->sortable()
```

or add a `relationCount`

```php
Columns\Text::make('orders')
    ->relationCount('orders')
    ->sortable()
```

**Comment by danharrin** [comment]
Hey, I think `relatedCount()` would do the trick, allowing custom formatted sorting would not be very performant.
